Assets/Scripts/TimeLoopController.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class TimeLoopController : MonoBehaviour, Timer
+{
+    [SerializeField] bool debugMode = false;
+    [SerializeField] float timeLimit = 10f;
+    private float timeCounter;
+
+    private bool resetting = false;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        timeCounter = timeLimit;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+        if (debugMode || resetting)
+        {
+            return;
+        }
+
+        if (timeCounter > 0) { 
+            // decrement time limit
+            timeCounter -= Time.deltaTime;
+
+            Debug.Log(timeCounter);
```

Comment:
Should probably remove or the console will print on every frame.

Reply:
fixed :)